src/update-mp-v1-ids-to-v2.ts

```diff
@@ -0,0 +1,174 @@
+/**
+ * 1. Run this codemod:
+ *
+ * In reaction...
+ *
+ *    ```bash
+ *    $ yarn codemod update-mp-v1-ids-to-v2
+ *    ```
+ *
+ * 5. Find TODOs added by codemod and update where necessary
+ */
+
+import { Transform, TaggedTemplateExpression, Identifier } from "jscodeshift"
+import {
+  parse as parseGraphQL,
+  print as printGraphQL,
+  visit,
+  NameNode,
+  FragmentDefinitionNode,
+  DefinitionNode,
+  ObjectTypeDefinitionNode,
+} from "graphql"
+import prettier from "prettier"
+import { ASTNode } from "recast"
+import { schema as promisedSchema } from "./helpers/getSchema"
+import { fragmentOnNonCompositeErrorMessage } from "graphql/validation/rules/FragmentsOnCompositeTypes"
+
+/**
+ * Follows the selection chain in a GQL operation and returns the _parent_ type
+ * from the related schema
+ */
+const walkSelections = (
+  selections: Array<string | number> | null,
+  operationChain: DefinitionNode,
+  schemaPosition: any
+): DefinitionNode => {
+  if (selections !== null && selections.length > 2) {
+    let selection = selections.shift()!
+    let definition = operationChain[selection] as DefinitionNode
+    let newSchemaPosition = schemaPosition
+    if (typeof selection === "number") {
+      newSchemaPosition = schemaPosition.fields.find(
+        field => definition.name.value === field.name.value
+      )
+    }
+    return walkSelections(selections, definition, newSchemaPosition)
+  }
+  return schemaPosition
+}
+
+const transform: Transform = async (file, api, _options) => {
+  if (!file.source.includes("graphql`")) {
+    return
+  }
+  const j = api.jscodeshift
+  const collection = j(file.source)
+  const schema = await promisedSchema
+
+  collection
+    .find(TaggedTemplateExpression, node => {
+      const tag = node.tag
+      return Identifier.check(tag) && tag.name === "graphql"
+    })
+    .forEach(path => {
+      const templateElement = path.node.quasi.quasis[0]
+      const graphqlDoc = parseGraphQL(templateElement.value.raw)
+
+      const newGraphQLDoc = visit(graphqlDoc, {
+        enter(node, key, parent, path, ancestors) {
+          // let p = Array.isArray(parent) ? parent[0] : parent
+          if (
+            node.kind === "Name" &&
+            parent &&
+            parent.kind == "Field" &&
+            (node.value === "id" ||
+              node.value === "_id" ||
+              node.value === "__id")
+          ) {
+            if (node.value === "__id") {
+              node.value = "id"
```

Comment:
I believe you’re supposed to return a new object, rather than mutate the original one, but YMMV 🤷‍♂